hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKey.java

```diff
@@ -96,12 +96,19 @@ public void serialize(@NonNull final SerializableDataOutputStream out) throws IO
 
     @Override
     public void serialize(@NonNull final ByteBuffer byteBuffer) throws IOException {
+        serializeReturningWrittenBytes(byteBuffer);
+    }
+
+    public int serializeReturningWrittenBytes(@NonNull ByteBuffer byteBuffer) throws IOException {
+        int initPos = byteBuffer.position();
         final var output = BufferedData.wrap(byteBuffer);
-        output.skip(4);
+        output.skip(Integer.BYTES);
```

Comment:
This information about serialized length doesn't seem to be used anywhere, it's redundant. At the same time, calculating the length makes code too complicated. Let's drop it from the serialized form and leave only essential part (what the codec actually writes).

Reply:
As of now, we still need length in deserialize:
```
    @Override
    public void deserialize(@NonNull final ByteBuffer byteBuffer, int ignored) throws IOException {
        final var buff = BufferedData.wrap(byteBuffer);
        final var len = buff.readInt();
        final var pos = buff.position();
        final var oldLimit = buff.limit();
        buff.limit(pos + len);
        key = codec.parse(buff);
        buff.limit(oldLimit);
    }
```
In the future we may get rid of it, but as of now it's still required